encoder_decoder.go

```diff
@@ -8,16 +8,18 @@ type encoder interface {
 	encode(pe packetEncoder) error
 }
 
+type encodeFunc func(pe packetEncoder) error
+
 // Encode takes an Encoder and turns it into bytes.
-func encode(in encoder) ([]byte, error) {
-	if in == nil {
+func encode(ef encodeFunc) ([]byte, error) {
```

Comment:
I don't understand this change at all, what was wrong with taking an interface here?